src/main/java/org/testeditor/web/xtext/index/resources/GlobalScopeResourceWithSeparateContextResourceSet.xtend

```diff
@@ -0,0 +1,190 @@
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import org.eclipse.emf.common.util.URI
+import org.eclipse.emf.ecore.EPackage
+import org.eclipse.emf.ecore.EReference
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl
+import org.eclipse.xtext.resource.impl.ResourceSetBasedResourceDescriptions
+import org.eclipse.xtext.scoping.IGlobalScopeProvider
+import org.eclipse.xtext.util.StringInputStream
+import org.slf4j.LoggerFactory
+
+@Path("/xtext/index/global-scope")
+@Produces(MediaType.APPLICATION_JSON)
+class GlobalScopeResourceWithSeparateContextResourceSet implements GlobalScopeResource {
+
+	protected static val logger = LoggerFactory.getLogger(GlobalScopeResourceWithSeparateContextResourceSet)
+
+	val IGlobalScopeProvider globalScopeProvider
+	val ResourceSetBasedResourceDescriptions index
+
+	new(IGlobalScopeProvider globalScopeProvider, ResourceSetBasedResourceDescriptions index) {
+		this.globalScopeProvider = globalScopeProvider
+		this.index = index
+	}
+
+	@POST // to allow context content to be passed as payload (instead of using query parameters)
+	@Consumes("text/plain")
+	@Produces("application/json")
+	override Response getScope(String context, @QueryParam("contentType") String contentType,
+		@QueryParam("contextURI") String contextURI, @QueryParam("reference") String eReferenceURIString) {
+		try {
+			val eReference = createEReference(eReferenceURIString)
+			val resource = createContextResource(context, contextURI, contentType)
+
+			logger.debug("Delegating to global scope provider='{}'", globalScopeProvider.class.simpleName)
+			val scope = globalScopeProvider.getScope(resource, eReference, null).allElements
+			logger.debug("Global scope provider returned the following elements='{}'", scope.map[name].join(','))
+
+			return Response.ok(scope.toList).build
+
+		} catch (GlobalScopeResourceException e) {
+			return Response.serverError.entity(e).build
+		}
+	}
+
+	private def createContextResource(String context, String contextURI, String contentType) {
+		logger.debug("Trying to retrieve or create context resource type='{}', URI='{}'", contentType, contextURI)
+		val resourceSet = new ResourceSetImpl // using empty resource set, since context will be used for this request only and is not part of the index
+		val resource = getOrCreateResource(resourceSet, contextURI, contentType)
+		if (!context.nullOrEmpty) {
+			loadResource(resource, context)
+		}
+		return resource
+	}
+
+	private def Resource getOrCreateResource(ResourceSet resourceSet, String contextURI, String contentType) {
+		var Resource result = null
+		
+		try {
+			val uri = URI.createURI(contextURI)
+			result = resourceSet.getResource(uri, false) ?: resourceSet.createResource(uri, contentType)
+		} catch (RuntimeException e) {
+			logger.error('Exception during creation of resource.', e)
```

Comment:
This try-catch if-else construct looks strange to me. I would leave the error handling to Dropwizard. If we really need that try-catch then return the result in the try block and do the error handling in the catch block. No need for the if-else block at all.